controllers/drplacementcontrol.go

```diff
@@ -558,7 +558,7 @@ func (d *DRPCInstance) validateRelocation(preferredCluster string) (string, erro
 func (d *DRPCInstance) readyToSwitchOver(homeCluster string, preferredCluster string) bool {
 	d.log.Info("Checking whether VRG is available", "cluster", homeCluster)
 
-	if isMetroAction(d.drPolicy, homeCluster, preferredCluster) {
+	if d.instance.Spec.Action != rmn.ActionRelocate && isMetroAction(d.drPolicy, homeCluster, preferredCluster) {
```

Comment:
Perhaps this check to check whether the action is relocate or not is not needed. Because this function "readyToSwitchOver" is invoked only in the relocate path (i.e. only when the action is Relocate in drpc spec).  hence the check will always be false.

In fact, with this check, the necessary verification to see if the cluster is fenced or not is not made at all. This can lead to DRPC trying to relocate to the preferred cluster even though it is fenced off.